src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1868,15 +1870,47 @@ public synchronized static void receivePeerIndexStatus(InetAddressAndPort endpoi
                 Index.Status status = Index.Status.valueOf(e.getValue());
                 indexStatus.put(keyspaceIndex, status);
             }
-
-            peerIndexStatus.put(endpoint, indexStatus);
+            
+            Map<String, Index.Status> oldStatus = peerIndexStatus.put(endpoint, indexStatus);
+            Map<String, Index.Status> delta = indexStatusDelta(oldStatus, indexStatus);
+            if (!delta.isEmpty())
+                logger.debug("Received updated index status for peer {}: {}", endpoint, formatIndexStatus(delta));
         }
         catch (Throwable e)
         {
             logger.warn("Unable to parse index status: {}", e.getMessage());
         }
     }
 
+    /**
+     * Returns a new map containing only the entries from newStatus that differ from corresponding entries in oldStatus.
+     */
+    private static @Nonnull  Map<String, Index.Status> indexStatusDelta(
+            @Nullable  Map<String, Index.Status> oldStatus,
+            @Nonnull Map<String, Index.Status> newStatus)
+    {
+        Map<String, Index.Status> delta = new HashMap<>();
+        for (Map.Entry<String, Index.Status> e : newStatus.entrySet())
+        {
+            if (oldStatus == null || e.getValue() != oldStatus.get(e.getKey()))
+                delta.put(e.getKey(), e.getValue());
```

Comment:
do we care about `Index.Status` that are only in `oldStatus` but not in `newStatus`?

Reply:
Removed statuses are also logged now.